deposit/sources/deposit.move

```diff
@@ -0,0 +1,46 @@
+address 0x1 {
+	module deposit {
+		use std::signer;
+
+		/// Deposit structure to transfer or fetch own balance
+		/// Bridge between Move and Substrate
+		struct Deposit has key {
+			destination: address,
+			amount: u128
+		}
+
+		/// Public constructor - means to instantiate `Deposit`
+		public fun new(destination: address, amount: u128): Deposit {
+			Deposit { destination, amount }
+		}
+
+		/// Transfer 'amount' from given `Deposit` to given 'destination'
+		/// Checks own balance prior to submitting the transaction
+		public fun do_deposit(account: signer, destination: address, amount: u128) acquires Deposit {
```

Comment:
this was the initial approach with borrowing, but I was unable to find a way to send reference to the function from calling code and it was simply not working on argument parsing. do you have fix suggestion for this? @Rqnsom

Reply:
I tested this locally half an hour ago - it should work like this:
```move
		/// Transfer 'amount' from given `Deposit` to given 'destination'
		/// Checks own balance prior to submitting the transaction
		public fun do_deposit(account: &signer, destination: address, amount: u128) acquires Deposit {
			// have to delete so it's not in cache / fresh
			let Deposit { destination: _, amount: _ } = move_from<Deposit>( signer::address_of(account));
			move_to(account, Deposit { destination, amount } );
		}
```


I was playing around the script (it works from what I tested):
```move
    fun transfer(me: signer, destination: address, amount: u128) {
        deposit::do_deposit(&me, destination, amount);

        assert!(deposit::check_balance(&me) == deposit::check_balance_of(std::signer::address_of(&me)), 0);

        deposit::do_deposit(&me, destination, 5);
    }
}
```

Reply:
I can try tomorrow to do some tweaking and then push a commit here - I have to leave now...

Reply:
For security and performance reasons `do_deposit` can be called once from script (enforced by relayed token on pallet side), which makes complete sense here to consume required non-copy argument to prevent possibility of compiling script which will never work on submission.

Reply:
> For security and performance reasons do_deposit can be called once from script

But that sounds like a huge limitation. We should be able to perform multiple transfers (and other operations with global operators which require a signer after a transfer is done) from the same address once somebody signs the transaction with such a script.

And if one wants to transfer funds between Substrate accounts, then that can be done without our pallet, right?

In EVM, it's possible to do multiple transfers within the same contract.

Reply:
It seems that multiple transfers within the same script should be one of our use cases. 
As I can see the potential performance reason in not doing that (as one bulk transfer is cheaper and faster than several transfers) but let's not decide for our users how they want to use our work :smile_cat:

Reply:
Please dont forget that we're working with substrate framework, which have it's own rules and methods. 
This part of implementation is for mapping native transfer method to move one, nothing more.
We can have additional item added into backlog for future work to implement [bulk transfers](https://polkadot.js.org/docs/api/cookbook/tx/#how-can-i-batch-transactions) alongside non-native asset management.
This piece remains as it is.